src/compiler/tracing.ts

```diff
@@ -122,16 +138,23 @@ namespace ts.tracing {
     }
     export function pop() {
         if (!traceFd) return;
-        Debug.assert(completeEvents.length > 0);
+        Debug.assert(canPop());
         const { phase, name, args, time } = completeEvents.pop()!;
         const dur = 1000 * timestamp() - time;
         writeEvent("X", phase, name, args, `"dur":${dur}`, time);
     }
+    export function canPop() {
+        return completeEvents.length > 0;
+    }
 
     function writeEvent(eventType: string, phase: Phase, name: string, args: object | undefined, extras?: string,
                        time: number = 1000 * timestamp()) {
         Debug.assert(traceFd);
         Debug.assert(fs);
+
+        // In server mode, there's no easy way to dump type information, so we drop events that would require it.
+        if (mode === Mode.Server && phase === Phase.CheckTypes) return;

```

Comment:
Is it completely useless without the type information?  -- Maybe it's better to just drop the information instead of not writing anything?

Reply:
1. Mostly, but not entirely, useless
 2. Server traces tend to cover longer time spans, so dropping events is helpful for keeping the sizes reasonable
 3. How would you identify the arguments that refer to type IDs?

Reply:
We can add them back later if we miss them.